### PR TITLE
feat(events-v2) Enable clicking on graph markers

### PR DESCRIPTION
**Pull request by markstory**
Clicking on a graph marker will now take you to an event in the selected timeslice. This requires an extra request to find an event in the slice. I've opted to use the `/latest` endpoint so that I don't have to deal with getting more a single result back, and we bias towards newer data in other event views.

Refs SEN-733

### CODE REVIEW

File: src/sentry/static/sentry/app/components/charts/utils.jsx [hunk 1]
```diff
@@ -56,6 +56,28 @@ export function getInterval(datetimeObj, highFidelity = false) {
...
 
+/**
+ * Convert an interval string into a number of seconds.
+ * This allows us to create end timestamps from starting ones
+ * enabling us to find events in narrow windows.
+ *
+ * @param {String} interval The interval to convert.
+ * @return {Integer}
+ */
+export function intervalToMilliseconds(interval) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/charts/utils.jsx]: We have this to hours: https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/utils.jsx#L246

File: src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx [hunk 2]
```diff
@@ -49,6 +59,55 @@ const getCurrentEventMarker = currentEvent => {
...
+
+  const url = `/organizations/${organization.slug}/events/latest/`;
+  let response;
+  try {
+    response = await api.requestPromise(url, {
+      method: 'GET',
+      query,
+    });
+  } catch (e) {
+    // Do nothing, user could have clicked on a blank space.
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx]: Thoughts (in general) about `console.error` in catch blocks? Sometimes they can end up swallowing errors in development and testing too.

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/charts/utils.jsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/charts/utils.jsx]: I can move this to live with its friend. I wonder if both shouldn't live in utils/dates.jsx?

File: src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx]: I think it makes sense when the failure is not expected to happen. I'm certain that users will click in blank spaces causing 404. I don't want to be notified in sentry when that happens though.

File: src/sentry/static/sentry/app/components/charts/utils.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/components/charts/utils.jsx]: I personally find the util functions hard to discover when they are grouped together in the same files, but moving them into `utils/dates` is an improvement 👍

File: src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx (hunk 2 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationEventsV2/modalLineGraph.jsx]: `console.error` won't log to sentry, I'm mainly thinking about during dev/tests (e.g. in tests if you don't mock this endpoint, the test will pass, maybe not applicable here, but in other places it's more relevant)

**Comment by markstory** [comment]
I've moved the interval parsing functions together and added the pin icon to the current event marker. The image is a bit wacky as echarts inverts our image when it positions it on the chart.

Screen Shot 2019-06-14 at 11 08 22 AM

**Comment by lynnagara** [comment]
Don't we know for sure that at least one event exists at a timestamp if a clickable marker has been rendered there? i.e. Could range just be 0 or 1 millisecond?

**Comment by markstory** [comment]
@lynnagara We can't use narrow ranges. As an example when you're looking at 14 days of data, each 'slice' in the graph represents 30 minutes of time, and we don't know where in the 30 minutes the event you're looking for is.

File: src/sentry/static/sentry/app/views/organizationEvents/utils/eventsRequest.jsx [hunk 3]
```diff
@@ -46,6 +46,7 @@ class EventsRequest extends React.PureComponent {
      * Absolute start date for query
      */
     start: PropTypes.instanceOf(Date),
+
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/organizationEvents/utils/eventsRequest.jsx]: nit: no need to touch this file

**Review by lynnagara — APPROVED** [review]

**Pull request merged**